### DISCUSSION TITLE
ant runinstall failed: CLI could not contact EJBCA instance

### DISCUSSION THREAD

**Discussion by lduczmal** [Keyfactor/ejbca-ce#680] [Q&A]
I'm trying to install ejbca-ce. I've already take a few attempts and each one finish on 'ant runinstall' which shows 'CLI could not contact EJBCA instance'. 
I'm sure that correct ports are open. I check whether jboss-cli.sh can connect on port 9990 to my wildfly instance on both localhost and FQDN - yes, can connect. 
I checked whether connection to DB (mariaDB) works correctly - yes, works correctly - I check it through MgmtWebGUI -> got to EJBCA datasource -> test connection -> test connection shows success. 
I discovered such info in /opt/wildfly/standalone/deployments/ejbca.ear.failed file:
\"WFLYCTL0412: Required services that are not installed:\" => [\"org.wildfly.ejb.remote\"] 
it looks like that my Wildfly instance hasn't all necessary components, but I install it using command published on official Keyfactor instruction: https://docs.keyfactor.com/ejbca/latest/wildfly-26#id-(8.3.2)WildFly26-DownloadandConfigureWildFlyUsingGalleon
Could you help me, please?

**Comment by primetomas** [comment]
You have to check the server.log. See https://docs.keyfactor.com/ejbca/latest/troubleshooting-guide

**Comment by lduczmal** [comment]
Hi, I followed instructions in troubleshooting-guide and attempted 'ant runinstall' once more. I suppose that most significant entry in server log is same as I have found in /opt/wildfly/standalone/deployments/ejbca.ear.failed file:
`ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "ejbca.ear")]) - failure description: {
    "WFLYCTL0412: Required services that are not installed:" => ["org.wildfly.ejb.remote"],
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [...`
Problem is that I don't know how to add missing service (org.wildfly.ejb.remote) to my Wildfly instance
[server.log](https://github.com/user-attachments/files/17282602/server.log)

**Comment by Alestor** [comment]
Put this into your standalone.xml under `<subsystem xmlns="urn:jboss:domain:ejb3:9.0">` between  `</timer-service>` and  `<thread-pools>`
```
           <remote cluster="ejb" connectors="http-remoting-connector" thread-pool-name="default">
                <channel-creation-options>
                    <option name="MAX_OUTBOUND_MESSAGES" value="1234" type="remoting"/>
                </channel-creation-options>
           </remote>
```
Ran into the same issue last week also using Galleon.

**Reply by lduczmal** [reply]
It didn't help.
I made another attempt of installing ejbca-ce on another machine (Ubuntu 23). In this attempt I chose to use full Wildfly (no Galleon installation) and I discover that although command 'ant -q clean deployear' finished with 'BUILD SUCCESSFUL' message, on Wildfly admin webconsole I see "The deployment of ejbca.ear is failed". And indeed in dir /opt/wildfly/standalone/deployments I found file  ejbca.ear.failed (I attached it to this post)
[ejbca.ear.failed](https://github.com/user-attachments/files/17444775/ejbca.ear.failed.txt)

**Reply by primetomas** [reply]
Looks like you don't have the database configured correctly as there is a MySQL error. It may be more clear MySQL errors earlier in the log. How did you configure your datasource? And are you sure the mysql/maridab driver deployed correctly?

**Reply by primetomas** [reply]
Also see discussion #688

**Reply by oliviermartin** [reply]
I had the same issue with installing WildFly 26 with Galleon. @Alestor suggestion fixed my issue.
